src/subscription/subscribe.js

```diff
@@ -106,44 +127,49 @@ function subscribeImpl(
   fieldResolver,
   subscribeFieldResolver
 ) {
-  const subscription = createSourceEventStream(
+  return createSourceEventStream(
     schema,
     document,
     rootValue,
     contextValue,
     variableValues,
     operationName,
     subscribeFieldResolver
-  );
-
-  // For each payload yielded from a subscription, map it over the normal
-  // GraphQL `execute` function, with `payload` as the rootValue.
-  // This implements the "MapSourceToResponseEvent" algorithm described in
-  // the GraphQL specification. The `execute` function provides the
-  // "ExecuteSubscriptionEvent" algorithm, as it is nearly identical to the
-  // "ExecuteQuery" algorithm, for which `execute` is also used.
-  return mapAsyncIterator(
-    subscription,
-    payload => execute(
-      schema,
-      document,
-      payload,
-      contextValue,
-      variableValues,
-      operationName,
-      fieldResolver
-    )
-  );
+  ).then(subscription => {
+    // For each payload yielded from a subscription, map it over the normal
+    // GraphQL `execute` function, with `payload` as the rootValue.
+    // This implements the "MapSourceToResponseEvent" algorithm described in
+    // the GraphQL specification. The `execute` function provides the
+    // "ExecuteSubscriptionEvent" algorithm, as it is nearly identical to the
+    // "ExecuteQuery" algorithm, for which `execute` is also used.
+    return mapAsyncIterator(
+      subscription,
+      payload => execute(
+        schema,
+        document,
+        payload,
+        contextValue,
+        variableValues,
+        operationName,
+        fieldResolver
+      ),
+      convertOrThrowError
+    );
+  }).catch(convertOrThrowError);
```

Comment:
You can provide `convertOrThrowError` as the 2nd argument to `then()` - which both reduces the number of wrapped promises, but also avoids calling `convertOrThrowError` two times in the case an error is thrown.